Remove warning from ActionView::Helpers::Tags::Translator

yuki24 opened the pull request:
This removes the following warning:

```
/GitHub/rails/actionview/lib/action_view/helpers/tags/translator.rb:19: warning: private attribute?
```

vipulnsward commented:
LGTM

egilburg commented:
Alternatively you can remove warning by using `protected` rather than `private`

IMO private attributes are nice in terms of a well-defined private API rather than free-floating instance vars, and also is better at detecting rename bugs during refactors (undefined variable warning is better than silently checking a now-nil instance var)

rafaelfranca commented:
:+1: for protected, but it is weird I can't have private attributes.

kaspth commented:
I don't see why we can't use private attributes here?

What Ruby version are you using, @yuki24?

kaspth commented on actionview/lib/action_view/helpers/tags/translator.rb:
```diff
@@ -10,26 +10,24 @@ def initialize(object, object_name, method_and_value, scope:)
 
         private
 
-        attr_reader :object_name, :method_and_value, :scope, :model
```
According to the warning this was the line that gave the warning.

I don't understand the error message (private attribute?), but perhaps the issue is with some of the definitions here.
Could you try putting each of them in separate declarations and post how many lines give warnings?

yuki24 committed:
Remove warning from ActionView::Helpers::Tags::Translator

This removes the following warning:

    /GitHub/rails/actionview/lib/action_view/helpers/tags/translator.rb:19: warning: private attribute?

yuki24 commented:
updated commit with `protected`. Thanks @egilburg!

@rafaelfranca @kaspth I'm not going to talk about private attrs here, but [Ruby has been giving this warning since 2007](https://github.com/ruby/ruby/blame/26a29ffbed868fd7cde8ccee74d70935fb1139b2/vm_method.c#L888-L910) and there's also [a discussion around it on twitter](https://twitter.com/stevenharman/status/489225805399326720). cc/ @tenderlove @ko1
